tensorflow/python/eager/forwardprop.py

```diff
@@ -140,6 +140,40 @@ def _jvp_helper(op_name, attr_tuple, inputs, outputs, tangents):
     return output_tangents
 
 
+def _jvp_helper_wrapper(
+  op_name, attr_tuple, inputs, outputs, tangents, batch_size
+):
+  """Computes a batch of Jacobian-vector product for an op.
+
+  Args:
+    op_name: A string, the type of operation being executed.
+    attr_tuple: Attributes of the operation.
+    inputs: A flat list of input Tensors to the operation.
+    outputs: A flat list of output Tensors from the operation.
+    tangents: A flat list of Tensors, same shape as `[batch_size] + input_shape`.
+
+  Returns:
+    A flat list of tangents corresponding to `outputs`.
+  """
+  if batch_size:
+    for primal, tangent in zip(inputs, tangents):
+      if tangent.shape != array_ops.concat([batch_size], primal.shape, 0):
```

Comment:
You might want to add a test for this error message if you leave it in. `self.assertRaises` or `self.assertRaisesRegexp` have a bunch of uses in the codebase.

The concat as it is doesn't look correct. You could do something like `tangent.shape.is_compatible_with([None] + primal.shape)`. It's a slightly looser check since it doesn't use batch_size, but then you don't have to worry about it being a tensor.

Reply:
That's way better. It would also allow for a list of primals to have different `batch_size`. We might as well have a bool `use_batch` instead of batch size then.